pkg/adapter/awssnssource/adapter.go

```diff
@@ -95,31 +94,16 @@ const (
 
 // Start implements adapter.Adapter.
 func (a *adapter) Start(ctx context.Context) error {
-	// ctx gets canceled to stop goroutines
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
-	// handle stop signals
-	go func() {
-		<-ctx.Done()
-		a.logger.Info("Shutdown signal received. Terminating")
-		cancel()
```

Comment:
This call to `cancel()` did absolutely nothing, since `ctx` is already "done" 2 lines above.